lib/term.js

```diff
@@ -2532,16 +2532,15 @@ Term.prototype.circle = function(center, radius, options) {
 
   return term;
 }
-Term.prototype.distance = function(geometry, options) {
+Term.prototype.distance = function(geometry, geometry2, options) {
   var self = this;
-  if (self._fastArityRange(arguments.length, 1, 2) === false) {
-    var _len = arguments.length;var _args = new Array(_len); for(var _i = 0; _i < _len; _i++) {_args[_i] = arguments[_i];}
-    self._arityRange(_args, 1, 2, 'distance', self);
-  }
+
+  self._noPrefix(self, 'distance');
   var term = new Term(self._r);
   term._query.push(termTypes.DISTANCE);
-  var args = [self, new Term(self._r).expr(geometry)];
+  var args = [new Term(self._r).expr(geometry), new Term(self._r).expr(geometry2)];
```

Comment:
geometry.distance(otherGeometry) is still a valid syntax, so you need to add a few if statements here